app/controllers/likes_controller.rb

```diff
@@ -1,32 +1,36 @@
 class LikesController < ApplicationController
   before_action :find_like
-  
+
   def create
     @like = @post.likes.new(user_id: current_user.id)
-    if @like.save
-      flash[:success] = "You liked a post."
-      redirect_to request.referrer
-    else
-      flash[:alert] = "You cannot like this post."
-      redirect_back
+    respond_to do |format|
+      if @like.save
+       format.js
+      else
+        flash[:alert] = "You cannot like this post."
+        redirect_back
+      end
```

Comment:
This doesn't look correct. format.js or format.html should be outside of the conditions